### PR TITLE
Book details

### PR DESCRIPTION
**Pull request by fededwolox**
## Summary ##

https://trello.com/c/QlPzgRS9/17-n-61-detalle-del-libro
https://trello.com/c/XCarstbr/18-n-62-tabla-de-comentarios
https://trello.com/c/gHmiG3uy/22-n-74-extension-para-botones-con-borde-y-gradiente

Book Detail with commentaries

### iPhone 5s

iphone5s

### iPhone X/XS/XR

iphoneXS

### CODE REVIEW

File: WBooks/BarControllers/NavBarController.swift [hunk 1]
```diff
@@ -11,9 +11,6 @@ import WolmoCore
 
 class NavBarController: UINavigationController {
     override func viewDidLoad() {
-        
-        super.viewDidLoad()
```
Comment (carcos-wolox) [inline WBooks/BarControllers/NavBarController.swift:15]: You shouldn't delete this.  All the functions that are about viewController's life cycle should have the `super` function call

File: WBooks/Extensions/UIButtonExtension.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            layer.cornerRadius = borderLineRadius
+        }
+        get {
+            return layer.cornerRadius
+        }
+    }
+    
+    func setBlueGradient() {
+        let colors = [
+            UIColor(red: 0.0, green: 0.68, blue: 0.93, alpha: 1),
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIButtonExtension.swift]: Add this colors to a `UIColor` extension

File: WBooks/Extensions/UIButtonExtension.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            UIColor(red: 0.22, green: 0.8, blue: 0.8, alpha: 1)]
+        
+        self.gradient = ViewGradient(colors: colors, direction: .leftToRight)
+}
+    
+    func setGreyGradient() {
+        let colors = [UIColor(red: 0.79, green: 0.79, blue: 0.79, alpha: 1),
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIButtonExtension.swift]: Same

File: WBooks/Extensions/UIImageViewExtension.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import UIKit
+
+extension UIImageView {
+    func loadImageUsingCache(withUrl urlString: String) {
+        image = UIImage.bookCover
+        guard let url = URL(string: urlString) else {
+            return
+        }
+        URLSession.shared.dataTask(with: url, completionHandler: { (data, _, error) in
+            if error != nil {
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIImageViewExtension.swift]: It's ok, but you can use `guard` here, it's more readable and simple :D

File: WBooks/Extensions/UIImageViewExtension.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//  WBooks
+//
+//  Created by Federico Agustin Diaz on 08/08/2019.
+//  Copyright © 2019 Wolox. All rights reserved.
+//
+
+import UIKit
+
+extension UIImageView {
+    func loadImageUsingCache(withUrl urlString: String) {
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIImageViewExtension.swift:12]: So nice this 👍

File: WBooks/Screens/BookDetail/BookCommentCellView.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    @IBOutlet weak var userName: UILabel!
+    @IBOutlet weak var userComment: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        contentView.backgroundColor = .white
+        backgroundColor = .clear
+        selectionStyle = .blue
+        userImage.layer.cornerRadius = userImage.frame.size.width/2
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookCommentCellView.swift]: Also, you can set this in userImage's `didSet` 👍

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = "BOOK_DETAIL".localized()
+        
+        configureTableView()
+        
+        initBookDetailTableViewModel()
+        
+        let backButton = UIBarButtonItem(image: UIImage(named: "Back"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(backButtonPressed))
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: Add this barButtonItem to a `UIBarButtonItem` extension

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        navigationItem.leftBarButtonItems = [backButton]
+    }
+    
+    override func viewDidLayoutSubviews() {
+        _detailHeaderView.setup(with: bookViewModel)
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: missing `super.viewDidLayoutSubviews()`

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        }
+        
+        bookDetailViewModel.loadComments(for: bookViewModel)
+    }
+    
+    private func configureTableView() {
+        _view.detailTable.delegate = self
+        _view.detailTable.dataSource = self
+        _view.configureDetailTableView()
+        let commentBookNib = UINib.init(nibName: "BookCommentCellView", bundle: nil)
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: Can you use wolmocore for his?

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return bookDetailViewModel.numberOfCells
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "BookCommentCellView", for: indexPath) as? BookCommentCellView else {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: Can you use wolmocore for his?

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+class BookDetailFullViewController: UIViewController {
+    
+    private let _view: BookDetailFullView = BookDetailFullView.loadFromNib()!
+    private let _detailHeaderView: BookDetailView = BookDetailView.loadFromNib()!
+    
+    lazy var bookDetailViewModel: BookDetailFullViewModel = {
+        return BookDetailFullViewModel()
+    }()
+    
+    var bookViewModel: BookDetails!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: This should me on the viewModel. Also, don't use `viewModel` if the object is not a viewModel because it's confusing

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    
+    convenience init(with bookViewModel: BookDetails) {
+        self.init()
+        self.bookViewModel = bookViewModel
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: `bookDetailViewModel` should have the bookDetails object, so in this init you can get the `BookDetailFullViewModel` from the parent

File: WBooks/Screens/BookDetail/BookDetailFullViewModel.swift [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        
+        let successComments: ([BookComment]) -> Void = { (comments) in
+            self.commentsViewModels = comments
+        }
+        
+        let failureComments: (Error) -> Void = { (error) in
+            self.showErrorAlertClosure?(error)
+        }
+        
+        WBNetworkManager.manager.getBookComments(book: bookView.book, onSuccess: successComments, onError: failureComments)
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewModel.swift]: You can have an onError closure in `loadComments` so you don't need to have `showErrorAlertClosure` in the viewModel.
```javascript
func  loadComments(onErrorComments: @escaping (Error) -> Void) {
    ...
    WBNetworkManager.manager.getBookComments(book: bookView.book, onSuccess: successComments, onError: onErrorComments)
}
```

File: WBooks/Screens/BookDetail/BookDetailFullViewModel.swift [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        
+        let successRent: (BookRent) -> Void = { (rent) in
+            self.showAlertClosure?("BOOK_RESERVED".localized())
+        }
+        
+        let failureRent: (Error) -> Void = { (error) in
+            self.showErrorAlertClosure?(error)
+        }
+        
+        WBNetworkManager.manager.rentBook(book: book.book, onSuccess: successRent, onError: failureRent)
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewModel.swift]: Same for onSuccess and onError

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        didSet {
+            wishlistButton.layer.cornerRadius = 20
+            wishlistButton.layer.borderWidth = 1
+            wishlistButton.layer.borderColor = UIColor.blueBorder().cgColor
+            wishlistButton.setTitleColor(.blueBorder(), for: .normal)
+        }
+    }
+    @IBOutlet weak var rentButton: UIButton! {
+        didSet {
+            rentButton.layer.cornerRadius = 20
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: If you are using the same `cornerRadius` for multiple buttons/views, you could add it to a global constant

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+            rentButton.layer.borderWidth = 1        }
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: same for borderWidth

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    @IBOutlet weak var bookImage: UIImageView!
+    
+    weak var delegate: DetailBookDelegate?
+    
+    func setup(with bookViewModel: BookDetails) {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: don't use `viewModel` if it's just a model

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+        bookAuthor.text = bookViewModel.bookAuthor
+        bookAvaiability.text = bookViewModel.bookStatus.bookStatusText()
+        bookAuthor.text = bookViewModel.bookAuthor
+        bookYear.text = bookViewModel.bookYear
+        bookGenre.text = bookViewModel.bookGenre
+        if bookViewModel.bookStatus.isBookAvailable() {
+            bookAvaiability.textColor = .green
+            rentButton.layer.borderColor = UIColor.blueBorder().cgColor
+            rentButton.setBlueGradient()
+            rentButton.clipsToBounds = true
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: `clipsToBound` is always true, so you can set this in `didSet`

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+        } else {
+            bookAvaiability.textColor = .red
+            rentButton.layer.borderColor = UIColor.greyButton().cgColor
+            rentButton.setGreyGradient()
+            rentButton.clipsToBounds = true
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: same

File: WBooks/Screens/BookLibrary/BookLibraryController.swift [hunk 20]
```diff
@@ -90,7 +93,12 @@ extension BookLibraryController: UITableViewDataSource {
 extension BookLibraryController: UITableViewDelegate {
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        cellSelected = tableView.cellForRow(at: indexPath) as? BookCell
+        let rectOfCell = tableView.rectForRow(at: indexPath)
+        rectOfCellSelected = tableView.convert(rectOfCell, to: tableView.superview)
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryController.swift]: Why do you need this?

File: WBooks/Screens/BookDetail/BookDetailView.swift [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+import UIKit
+import WolmoCore
+
+protocol DetailBookDelegate: class {
+    func rentBook()
+}
+
+class BookDetailView: UITableViewCell, NibLoadable {
+    
+    @IBOutlet weak var bookTitle: UILabel!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailView.swift]: In the screencast, this title is not showing the whole text. Make it take 2 lines

File: WBooks/Screens/BookLibrary/BookLibraryViewModel.swift [hunk 22]
```diff
@@ -10,28 +10,49 @@ import UIKit
...
+    var bookLibraryModel: [Book] = [] {
         didSet {
-            self.reloadViewClosure?()
+            reloadViewClosure?()
         }
     }
     
-    var numberOfCells: Int {
-        return libraryItems.count
+    private var bookDetailsModel: [BookDetails] = [] {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryViewModel.swift]: Whats the difference between `bookDetailsModel` and `bookLibraryModel`  ?  If i'm ok, `BookDetails` as a `Book` property. Make sure this is not duplicated.

File: WBooks/Screens/BookLibrary/BookLibraryViewModel.swift [hunk 23]
```diff
@@ -10,28 +10,49 @@ import UIKit
...
     
     var reloadViewClosure: (() -> Void)?
     
-    func getCellBook(at indexPath: IndexPath) -> Book {
-        return libraryItems[indexPath.row]
+    func getCellBookLibrary(at indexPath: IndexPath) -> Book {
+        return bookLibraryModel[indexPath.row]
+    }
+    
+    func getCellBookDetail(at indexPath: IndexPath) -> BookDetails {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryViewModel.swift]: I'm not sure if you are using this function

File: WBooks/Screens/BookLibrary/BookLibraryViewModel.swift [hunk 24]
```diff
@@ -10,28 +10,49 @@ import UIKit
...
+        return bookDetailsModel[indexPath.row]
     }
     
     func loadBooks() {
-        WBNetworkManager.manager.fetchBooks(onSuccess: { (books) in
-            self.libraryItems = books
-        }, onError: { (error) in
+        let successBooks: ([Book]) -> Void = { (books) in
+            self.bookLibraryModel = books
+            self.bookDetailsModel = self.bookLibraryModel.map { BookDetails(book: $0) }
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryViewModel.swift]: This is why I think `bookLibraryModel` and  `bookDetailsModel` are the same. You could have just `bookDetailsModel`

File: WBooks/WBNetworkManger.swift [hunk 25]
```diff
@@ -35,10 +35,80 @@ class WBNetworkManager: NSObject {
...
+    }
+    
+    public func rentBook(book: Book, onSuccess: @escaping (BookRent) -> Void, onError: @escaping (Error) -> Void) {
+        
+        let userId = 7
+        
+        let url = URL(string: "https://swift-training-backend.herokuapp.com/users/\(userId)/rents")!
+        
+        let params: [String: Any] = ["userID": userId,
+            "bookID": book.id,
```
Comment (carcos-wolox) [inline WBooks/WBNetworkManger.swift]: indentation

File: WBooks/WBNetworkManger.swift [hunk 26]
```diff
@@ -35,10 +35,80 @@ class WBNetworkManager: NSObject {
...
     }
     
+    private func firstDate() -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.locale = Locale.current
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        return dateFormatter.string(from: Date())
+    }
+    
+    private func secondDate() -> String {
```
Comment (carcos-wolox) [inline WBooks/WBNetworkManger.swift]: You could have a function in `UIDate`  extension, that gets a Date and returns a String with the desired format, and just call it with `Date()` and ` Calendar.current.date(byAdding: .day, value: 1, to: Date())!`

**Review by carcos-wolox — CHANGES_REQUESTED** [review]
Some changes

File: WBooks/Constants.swift [hunk 27]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Federico Agustin Diaz on 14/08/2019.
+//  Copyright © 2019 Wolox. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+struct Constants {
+    static var cornerRadiusBig: CGFloat {
```
Comment (carcos-wolox) [inline WBooks/Constants.swift]: Doesn't need to be computed properties. Could be just let constants

File: WBooks/Constants.swift [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+        return 20
+    }
+    static var cornerRadiusSmall: CGFloat {
+        return 5
```
Comment (carcos-wolox) [inline WBooks/Constants.swift]: same

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+class BookDetailFullViewController: UIViewController {
+    
+    private let _view: BookDetailFullView = BookDetailFullView.loadFromNib()!
+    private let _detailHeaderView: BookDetailView = BookDetailView.loadFromNib()!
+    
+    lazy var bookDetailViewModel: BookDetailFullViewModel = {
+        return BookDetailFullViewModel()
+    }()
+    
+    var bookModel: Book!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: this should be a ViewModel property, not in the controller

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+import UIKit
+import WolmoCore
+
+class BookDetailFullViewController: UIViewController {
+    
+    private let _view: BookDetailFullView = BookDetailFullView.loadFromNib()!
+    private let _detailHeaderView: BookDetailView = BookDetailView.loadFromNib()!
+    
+    lazy var bookDetailViewModel: BookDetailFullViewModel = {
+        return BookDetailFullViewModel()
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: Get this viewModel from the init

File: WBooks/Extensions/UIColorExtension.swift [hunk 31]
```diff
@@ -12,4 +12,25 @@ extension UIColor {
...
+        return UIColor(red: 0.79, green: 0.79, blue: 0.79, alpha: 1)
+    }
+    class func blueFirstGradient() -> UIColor {
+        return UIColor(red: 0.0, green: 0.68, blue: 0.93, alpha: 1)
+    }
+    class func blueSecondGradient() -> UIColor {
+        return UIColor(red: 0.22, green: 0.8, blue: 0.8, alpha: 1)
+    }
+    class func greyFirstGradient() -> UIColor {
+        return UIColor(red: 0.79, green: 0.79, blue: 0.79, alpha: 1)
```
Comment (carcos-wolox) [inline WBooks/Extensions/UIColorExtension.swift]: This color is the same as `greyButton`

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 32]
```diff
@@ -0,0 +1,119 @@
...
+
+import UIKit
+import WolmoCore
+
+class BookDetailFullViewController: UIViewController {
+    
+    private let _view: BookDetailFullView = BookDetailFullView.loadFromNib()!
+    private let _detailHeaderView: BookDetailView = BookDetailView.loadFromNib()!
+    
+    var bookDetailViewModel: BookDetailFullViewModel!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: This could be private

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 33]
```diff
@@ -0,0 +1,119 @@
...
+    // MARK: - Private
+    private func initBookDetailTableViewModel() {
+        
+        bookDetailViewModel.reloadViewClosure = { [weak self] () in
+            DispatchQueue.main.async {
+                self?._view.detailTable.reloadData()
+            }
+        }
+        
+        bookDetailViewModel.loadComments(for: bookDetailViewModel.bookModel, onErrorComments: { [weak self] (error) in
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: You can also get the reloadViewClosure as `loadComments` param

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift [hunk 34]
```diff
@@ -0,0 +1,119 @@
...
+            }
+        })
+    }
+    
+    private func configureTableView() {
+        _view.detailTable.delegate = self
+        _view.detailTable.dataSource = self
+        _view.configureDetailTableView()
+        let commentBookNib = UINib.init(nibName: "BookCommentCellView", bundle: nil)
+        _view.detailTable.register(commentBookNib, forCellReuseIdentifier: "BookCommentCellView")
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: Can you use wolmoCore for registering the cell?

File: WBooks/Screens/BookDetail/BookDetailFullViewModel.swift [hunk 35]
```diff
@@ -0,0 +1,63 @@
...
+    init(with book: Book) {
+        bookModel = book
+    }
+    
+    var numberOfCells: Int {
+        return commentsViewModels.count
+    }
+    
+    var reloadViewClosure: (() -> Void)?
+    var showErrorAlertClosure: ((Error) -> Void)?
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewModel.swift]: You are not assigning a value for this property, so you can delete it

File: WBooks/Screens/BookDetail/BookDetailFullViewModel.swift [hunk 36]
```diff
@@ -0,0 +1,63 @@
...
+    func loadComments(for bookView: Book, onErrorComments: @escaping (Error) -> Void) {
+        
+        let successComments: ([BookComment]) -> Void = { (comments) in
+            self.commentsViewModels = comments
+        }
+        
+        WBNetworkManager.manager.getBookComments(book: bookView, onSuccess: successComments, onError: onErrorComments)
+    }
+    
+    func rentBook(book: Book) {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewModel.swift]: Same as `loadComments`, you can get `successRent` and `failureRent` as function params

File: WBooks/Screens/BookDetail/BookDetailFullViewModel.swift [hunk 37]
```diff
@@ -0,0 +1,63 @@
...
+        
+        let successRent: (BookRent) -> Void = { (rent) in
+            self.showAlertClosure?("BOOK_RESERVED".localized())
+        }
+        
+        let failureRent: (Error) -> Void = { (error) in
```
Comment (carcos-wolox) [inline WBooks/Screens/BookDetail/BookDetailFullViewModel.swift]: If you are not adding anything else to the closure, you don't need to create another closure that executes the previous closure

File: WBooks/Screens/BookLibrary/BookLibraryController.swift [hunk 38]
```diff
@@ -22,14 +23,17 @@ class BookLibraryController: UIViewController {
         fatalError("init(nibName:bundle:) has not been implemented")
     }
     
-    var libraryViewModel: BookLibraryViewModel = {
+    var bookDetailsModel: BookLibraryViewModel = {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryController.swift]: This doesn't need to be a computed property. you can have it like this: `let bookLibraryViewModel = BookLibraryViewModel()`

File: WBooks/Screens/BookLibrary/BookLibraryController.swift [hunk 39]
```diff
@@ -22,14 +23,17 @@ class BookLibraryController: UIViewController {
...
         return BookLibraryViewModel()
     }()
     
     init() {
         super.init(nibName: .none, bundle: .none)
     }
     
+    var cellSelected: BookCell!
+    var rectOfCellSelected: CGRect!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryController.swift]: Delete this property if you are not using it

File: WBooks/Screens/BookLibrary/BookLibraryController.swift [hunk 40]
```diff
@@ -22,14 +23,17 @@ class BookLibraryController: UIViewController {
         fatalError("init(nibName:bundle:) has not been implemented")
     }
     
-    var libraryViewModel: BookLibraryViewModel = {
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryController.swift]: Why did you change the name? it's confusing because its a libraryViewModel, not a single bookDetailsViewModel

File: WBooks/Screens/BookLibrary/BookLibraryController.swift [hunk 41]
```diff
@@ -22,14 +23,17 @@ class BookLibraryController: UIViewController {
...
+    var bookDetailsModel: BookLibraryViewModel = {
         return BookLibraryViewModel()
     }()
     
     init() {
         super.init(nibName: .none, bundle: .none)
     }
     
+    var cellSelected: BookCell!
```
Comment (carcos-wolox) [inline WBooks/Screens/BookLibrary/BookLibraryController.swift]: Delete this property if you are not using it

**Review by carcos-wolox — CHANGES_REQUESTED** [review]
Few comments 👍

File: WBooks/Screens/BookDetail/BookDetailFullViewController.swift (hunk 34 above)
Comment (fededwolox) [reply WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: It gives me an error, which doesnt happen with BookCell
Cannot invoke 'register' with an argument list of type '(cell: BookCommentCellView.Type)'

Comment (carcos-wolox) [reply WBooks/Screens/BookDetail/BookDetailFullViewController.swift]: `BookCommentCellView`  needs to be `Nibloadable`

**Review by carcos-wolox — APPROVED** [review]
Nice!! good job. Please check `BookCommentCellView` register cell 👍 👍
